### ISSUE TITLE
VSCode's "Go To Definition" should jump to the actual TS source

### ISSUE THREAD

**Issue by johannes-gehrs** [hashicorp/terraform-cdk#2881]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When jumping to source in VSCode in a project that works with CDKTF the editor jumps to the d.ts file rather than to the actual TS source. I believe this is becuase the compile option `"declarationMap": true` has not been set. Here's a quote from the TS [reference docs](https://www.typescriptlang.org/tsconfig#declarationMap):

> Generates a source map for .d.ts files which map back to the original .ts source file. This will allow editors such as VS Code to go to the original .ts file when using features like Go to Definition. You should strongly consider turning this on if you’re using project references.

And here's a quote from a [relevant Stack Overflow answer](https://stackoverflow.com/a/58461375/1430384):

> The IDE can make use of [declarationMap](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#new---declarationmap) option to provide better developer experience: It enables you to quickly navigate to the original sources, when you currently have a corresponding .d.ts file open and want to see its implementation

In summary, I believe it would be useful to enable this compiler feature, because it substantially enhances the developer experience - in particular when using CDKTF programmatically to do non-standard things.

### References

n/a

**Comment by ansgarm** [comment]
Sounds good! A PR that adds this would be very welcome, if you want to contribute! 🙂

**Comment by xiehan** [comment]
Fixed in #2889
